### ISSUE TITLE
What if we use using both the pretraining objective (e.g. Mask image prediction loss in MAE) and the NNA loss to pre-train a model?

### ISSUE THREAD

**Issue by GT9505** [ml-jku/MIM-Refiner#2]
Hi, I'm wondering why using the Nearest Neighbor Alignment (NNA) loss to finetune the pre-trained weights.

What if we use both the pretraining objective (e.g. Mask image prediction loss in MAE) and the NNA loss to pre-train a model? Is it possible to solve the problem of Fig 3(d) in the paper? Could the performance be higher than MIM-Refiner?

**Comment by BenediktAlkin** [comment]
Hi, there are works that try to combine these two objective in an end-to-end fashion (e.g. [CMAE](https://arxiv.org/abs/2207.13532) or our previous work [MAE-CT](https://arxiv.org/abs/2304.10520)).
The main problem is that MIM and ID objectives are very different.

MIM distributes attention equally among patches because it reconstructs all parts of the image which makes all patches equally important. It also benefits from high mask ratios (75%) and little augmentations (only RandomResizedCrop and horizontal flip).

ID on the contrary focuses attention onto objects as they are the most reliable way to find the positive sample for the ID task. ID is best when no masking (or little masking) is applied and also uses a lot of augmentations (ColorJitter, GaussianBlur, Solarization, Grayscale, local crops) which all worsen the performance of MIM.

We investigated an end-to-end training in [MAE-CT](https://arxiv.org/abs/2304.10520) but it was always less successful than fine-tuning with only the ID objective.

The most successful works that combine MIM and ID are iBOT and DINOv2. Other approaches can improve the performance of small models (up to ViT-B) by combining the two objectives, but on larger models (where MIM shines) they are worse and require much more compute.

**Comment by GT9505** [comment]
Got it! Thanks a lot.

I have another question. What is the main difference between MIM-Refiner and MAE-CT? Is it that MIM-Refiner uses multiple ID heads for the feature calibration of multi-stage while MAE-CT uses only one ID head for the feature  calibration of the last stage?

**Comment by BenediktAlkin** [comment]
Yes, the largest difference is that MIM-Refiner uses multiple ID heads which makes it compatible with all MIM models instead of only MAE. MAE-CT for example is worse with a H/14 than with a H/16 despite the H/14 being the model that uses more FLOPS. That is because the encoder/decoder imbalance is worse for the H/14 than it is with the H/16. MAE-CT would also not work with data2vec or any other MIM model as the representation in the last encoder layer of MIM models is terrible.
MAE has the best representation in the last layer of the common MIM models (as MAE has one of the "largest" decoders among MIM models).

**Comment by GT9505** [comment]
Thanks a lot!
